### PR TITLE
Handels XPath with absolute path correctly

### LINKED ISSUE #100: Xpath does not navigate from root when expected
### Bug Report or Feature Request (mark with an x)
- [x] bug report -> please search issues before submitting
- [ ] feature request

### Expected Behavior
```csharp
var xml = new XmlDocument();
xml.Load("<root><child name='Jerome'><property name='test'/></child></root>");
new Xambler(
   new Directives().
          .Xpath("/root/child[@name='Jerome']/property[@name='test']")
          .Remove()
          .Xpath("/root/child[@name='Jerome']
          .Add("property")
          .Attr("name", "test2")
    ).Apply(xml).OuterXml == "<root><child name='Jerome'><property name='test2'/></child></root>";
```

### Actual Behavior
Result is 
```xml
<root><child name='Jerome'/></root>
```

### Mention any other details that might be useful
Xpath starting with character '/' means, that it should navigate the path from root always. But it seems as if it navigates from the last position(?).

### PR DESCRIPTION
**Pull request by aveXcaesar**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)
#100 

### What is the new behavior?
The `XpathDirective()` is more robust an handles absolute XPath expressions by an given illegal cursor.
The `RemoveDirective()` returns cursors to the parents now.

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information
closes #100

### CODE REVIEW

**Comment by koeeenig** [comment]
@aveXcaesar please resolve the conflicts. After that we will start the review process. Thanks :)

**Comment by aveXcaesar** [comment]
@koeeenig fixed

File: src/Yaapii.Xml.Xambly/Directive/XpathDirective.cs [hunk 1]
```diff
@@ -88,7 +94,28 @@ public ICursor Exec(XmlNode dom, ICursor cursor, IStack stack)
             //{
             //    targets = this.Traditional(query, dom, cursor);
             //}
-            targets = this.Traditional(query, dom, cursor);
+
+            // CSA: This fork is only needed if the cursor points to an XmlNode which is not existing in the dom.
```
Comment (koeeenig) [inline src/Yaapii.Xml.Xambly/Directive/XpathDirective.cs]: @aveXcaesar I think infos like you wrote here should go into a comment section of the issue or as comment in the PR.
These comments look like personal notes on the learnings you made while working on this issue. 
Of course this are important values for everybody but i think the place were they are document is wrong.
I will remove them from the code.

Comment (aveXcaesar) [reply src/Yaapii.Xml.Xambly/Directive/XpathDirective.cs]: Next time I have to think about it again. This is a very complicated section and it takes much time to understand it.
You are right to clean up the code. But in this class I think we should keep it...

Comment (Meerownymous) [reply src/Yaapii.Xml.Xambly/Directive/XpathDirective.cs]: @avexcaesar this is a public repo, ypur personal username must not appear here!

**Pull request merged**